src/middle-tier/app.js

```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright (c) Microsoft. All rights reserved. Licensed under the MIT license. See full license in root of repo. -->
+ *
+ * This file is the main Node.js server file that defines the express middleware.
+ */
+
+if (process.env.NODE_ENV !== "production") {
+  require("dotenv").config();
+}
+import * as createError from "http-errors";
+import * as path from "path";
+import * as cookieParser from "cookie-parser";
+import * as logger from "morgan";
+import express from "express";
+import https from "https";
+import { getHttpsServerOptions } from "office-addin-dev-certs";
+import { getUserData } from "./msgraph-helper";
+import { validateJwt } from "./ssoauth-helper";
+
+/* global console, process, require, __dirname */
+
+const app = express();
+const port = process.env.API_PORT || "3000";
+
+app.set("port", port);
+
+// view engine setup
+app.set("views", path.join(__dirname, "views"));
+app.set("view engine", "pug");
+
+app.use(logger("dev"));
+app.use(express.json());
+app.use(express.urlencoded({ extended: false }));
+app.use(cookieParser());
+
+/* Turn off caching when developing */
+if (process.env.NODE_ENV !== "production") {
+  app.use(express.static(path.join(process.cwd(), "dist"), { etag: false }));
+
+  app.use(function (req, res, next) {
+    res.header("Cache-Control", "private, no-cache, no-store, must-revalidate");
+    res.header("Expires", "-1");
+    res.header("Pragma", "no-cache");
+    next();
+  });
+} else {
+  // In production mode, let static files be cached.
+  app.use(express.static(path.join(process.cwd(), "dist")));
+}
+
+const indexRouter = express.Router();
+indexRouter.get("/", function (req, res) {
+  res.render("/taskpane.html");
+});
+
+app.use("/", indexRouter);
+
+// Middle-tier API calls
+// listen for 'ping'
+app.get("/ping", function (req, res) {
```

Comment:
Is this required? Would be nice not to have any extra public APIs returning information to anyone who queries it.

Reply:
Or at least warn in the readme to remove/update this for production deployment.

Reply:
I'll add a comment in the code (I believe we delete the readme when "hydrating" the template).